src/screens/MessageScreen.tsx

```diff
@@ -87,31 +133,37 @@ export function MessageScreen({
     [userQueries],
   );
 
-  if (
-    isLoading ||
-    userQueries.some((query) => query.isLoading) ||
-    userLoading
-  ) {
+  const handleScroll = useCallback(
+    (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+      if (
+        event.nativeEvent.contentOffset.y +
+          event.nativeEvent.layoutMeasurement.height >=
+        event.nativeEvent.contentSize.height
+      ) {
+        if (recipientsListChunked.length > scrollIndex) {
+          // Add new users to view on scroll
+          setScrollIndex(scrollIndex + 1);
```

Comment:
This could prevent an extra render.
```suggestion
          setScrollIndex(currentIndex => currentIndex + 1);
```